### PR TITLE
Additional S2 support + new S2 boards (v2)

### PR DESCRIPTION
**Pull request by dpgeorge**
This is a rework of #7163 to tidy up the commits.

### CODE REVIEW

File: ports/esp32/main.c [hunk 1]
```diff
@@ -206,3 +220,9 @@ void *esp_native_code_commit(void *buf, size_t len, void *reloc) {
     memcpy(p, buf, len);
     return p;
 }
+
+#if CONFIG_IDF_TARGET_ESP32S3
+float __ieee754_sqrtf(float x) {
```
Comment (dpgeorge) [inline ports/esp32/main.c:225]: @UnexpectedMaker why is this needed?  surely it needs a proper implementation :)

Comment (UnexpectedMaker) [reply ports/esp32/main.c:225]: That was what `we` added when we did that video call a while back as something in the compiler was looking for it and it was falling over without it.

**Comment by UnexpectedMaker** [comment]
Thanks for re-jigging my PR commits :)

File: ports/esp32/main.c (hunk 1 above)
Comment (dpgeorge) [reply ports/esp32/main.c:225]: Oh, I don't remember that!  Do you have any S3 board definitions that compile?

Comment (UnexpectedMaker) [reply ports/esp32/main.c:225]: I'll need to test out my S3 board again today and I'll get back to you with files for a working S3 board.

Comment (UnexpectedMaker) [reply ports/esp32/main.c:225]: > Oh, I don't remember that! Do you have any S3 board definitions that compile?

I've not had a chance to look into this yet, sorry. Feel free to remove it for now as its S3 specific only.

Comment (dpgeorge) [reply ports/esp32/main.c]: Ok, removed this for now.
